### ISSUE TITLE
RetroArch crash upon hosting a netplay session [Android]

### ISSUE THREAD

**Issue by Keltron3030** [libretro/RetroArch#13658]
RetroArch crashes when hosting a netplay session after loading content.

The crash can be avoided by first hosting the netplay session and then launching the content thereafter.

### Steps to reproduce the bug

1. Launch any game (tested with FCEUmm, Genesis Plus GX, & Picodrive).
2. Host a netplay session.
3. RetroArch will become unresponsive on the menu and crash after approximately 1 minute.

This crash will occur consistently on Android builds as of 1.10.0. 

Using 1.9.14, the game content screen will return but Retroarch will freeze and crash after a few seconds.  However, sometimes the crash will not occur at all.

Using 1.9.13 no crash will occur.

- RetroArch: [1.10.0/5f6fc20]

- OS: Android 10 & 12 tested and crash confirmed on both.

**Comment by Keltron3030** [comment]
Toggling "Allow Pausing" on in settings>network will eliminate the crashing problem on 1.10.0

Thanks @Cthulhu-throwaway for the suggestion

**Comment by Cthulhu-throwaway** [comment]
@Keltron3030 Netplay content reload should fix this issue aswell. Could you test the currently nightly?

@LibretroAdmin not sure if the author is still around, but if not, I guess someone else could try enabling netplay with content already initialized on Android; if it's fixed, than this can be closed.

**Comment by Keltron3030** [comment]
I'll give it a try later and report, thanks.

**Comment by Clutz450** [comment]
@Cthulhu-throwaway Thank you for your reply in the other issue report.  I am continuing our discussion over here as requested.  

You had said over there (and please forgive me, I don't know how to quote from another issue):
"The crash you are experiencing may or may not be related to netplay; hard to say since you can't reproduce it other than that one time and you didn't provide/have a stacktrace of the crash."
How do I provide you with a stack trace of the crash?  I see the Logging options under settings but not sure which options need to be enabled.  I did have it crash on me again randomly so I'd like to see if I can make it happen again and get you that stacktrace.

I noticed another possible issue when comparing netplay with the current stable to the nightlie.  Using the stable and refreshing netplay hosts, the crc of the rom looks normal.  But on the nightlie, there are a bunch of "F"s in front of it.  I'm attaching screenshots so you can see what I mean.  Let me know if this is unrelated and you'd like for me to create a new issue.

So that's all I've done and noticed so far.  I'll get you that stacktrace as soon as I know how to create one and make it crash again.  Other then that, please let me know what else you'd like for me to try out and I'd be more than happy to do it.

![Screenshot_20220822-125026_RetroArch](https://user-images.githubusercontent.com/38346645/185979392-d5b2b4d2-329e-44e1-918c-3741ecb4224c.jpg)
![Screenshot_20220822-125008_RetroArch](https://user-images.githubusercontent.com/38346645/185979421-7e4362ad-50a2-46f7-ba3a-114c2183d5b8.jpg)
![Screenshot_20220822-123457_RetroArch](https://user-images.githubusercontent.com/38346645/185979436-e350a5ba-0e8a-46b0-b7a3-68a56cc33869.jpg)
![Screenshot_20220822-123440_RetroArch](https://user-images.githubusercontent.com/38346645/185979453-9294c939-0dea-46d5-bf0b-3da2784061fa.jpg)

**Comment by Cthulhu-throwaway** [comment]
@Clutz450 Your ARM v8 CPU is 64-bits, right? Modern Unix compilers will treat long as a 64-bits type on 64-bits platforms, while on Windows, long is still treated as a 32-bits type even on x86_64.

I'll take a look at the label code to see why the higher bits are 1, rather than 0.

**Comment by Clutz450** [comment]
I believe I am 64 bit but I am not sure.  My cell phone is a Samsung Galaxy S21 if that helps.

Glad to see you figured out the other issue.

**Comment by Cthulhu-throwaway** [comment]
> I believe I am 64 bit but I am not sure. My cell phone is a Samsung Galaxy S21 if that helps.
> 
> Glad to see you figured out the other issue.

All the reported CPUs here https://en.wikipedia.org/wiki/Samsung_Galaxy_S21 are ARM 64-bits, but it was pretty obvious you were running a 64-bits RetroArch because long is 64-bits.

**Comment by Clutz450** [comment]
So I updated to the latest nightlie again and I assumed it was going to be the same one that I posted above (since the build date was still the 23rd) but it was a different Git Version.  This latest one that I just downloaded is 85c44f9.  This latest one appeared to freeze for a bit when starting netplay but then ultimately connected. 
 Whereas the one I posted about above was c97bc0d and that would freeze every time (like I waited until my phone said that the app wasn't responded and asked me if I wanted to wait or to close it).

Anyway, what I am getting at is that I thought it was weird that latest stable I had the freeze, updated to latest nightly at the time and netplay worked but crcs were messed up, updated again to latest nightly the next day and crcs were correct but netplay froze again, then just now updated again to latest nightlie and everything seems to work now.

Bottom line is it appears with this latest nightlie (Git Version 85c44f9) both the crc issue and the netplay freezing issue are fixed.  I'm going to stop confusing myself by testing all different version of RetroArch and wait for you to reply back to see what exactly you may need me to do next.

**Comment by Keltron3030** [comment]
Still not working properly for me with the nightly. I start the game (Balloon Fight) using FCEUmm. Once I attempt to host a netplay session Retroarch becomes unresponsive.

**Comment by Cthulhu-throwaway** [comment]
> So I updated to the latest nightlie again and I assumed it was going to be the same one that I posted above (since the build date was still the 23rd) but it was a different Git Version. This latest one that I just downloaded is [85c44f9](https://github.com/libretro/RetroArch/commit/85c44f9245e5b6d9f157c9966f49e12c594efebe). This latest one appeared to freeze for a bit when starting netplay but then ultimately connected. Whereas the one I posted about above was [c97bc0d](https://github.com/libretro/RetroArch/commit/c97bc0ddbd029cd20468d7b0c707e0eb3ba66ad4) and that would freeze every time (like I waited until my phone said that the app wasn't responded and asked me if I wanted to wait or to close it).
> 
> Anyway, what I am getting at is that I thought it was weird that latest stable I had the freeze, updated to latest nightly at the time and netplay worked but crcs were messed up, updated again to latest nightly the next day and crcs were correct but netplay froze again, then just now updated again to latest nightlie and everything seems to work now.
> 
> Bottom line is it appears with this latest nightlie (Git Version [85c44f9](https://github.com/libretro/RetroArch/commit/85c44f9245e5b6d9f157c9966f49e12c594efebe)) both the crc issue and the netplay freezing issue are fixed. I'm going to stop confusing myself by testing all different version of RetroArch and wait for you to reply back to see what exactly you may need me to do next.

The short "freeze" is the netplay auto core updater updating your core in the background to ensure you always netplay at the latest version. Can be disabled by locking the core, but should be avoided for obvious reasons, unless you've some very specific reason(s) not to want a sync to the latest core version.

Achievements were deadlocking when reloading content for netplay, then it got fixed, then the fix got reverted because it broke something else. It should be fixed now with the aforementioned PR as you've experienced it yourself.

**Comment by Cthulhu-throwaway** [comment]
> Still not working properly for me with the nightly. I start the game (Balloon Fight) using FCEUmm. Once I attempt to host a netplay session Retroarch becomes unresponsive.

Take a look at my previous post.

**Comment by Keltron3030** [comment]
Yep, working good now. Thanks for your efforts 👍

**Comment by Keltron3030** [comment]
Resolved

**Comment by Keltron3030** [comment]
Resolved
